lib/pure/httpclient.nim

```diff
@@ -980,7 +980,7 @@ proc requestAux(client: HttpClient | AsyncHttpClient, url, httpMethod: string,
   var data: seq[string]
   if multipart != nil and multipart.content.len > 0:
     data = await client.format(multipart)
-  else:
+  elif not(httpMethod == "GET" and body.len == 0):
```

Comment:
Would HEAD/CONNECT/OPTIONS/TRACE also benefit from this? Also very minor note: `not` is an operator, having no space after it looks a little weird to me.

Reply:
Added space as requested.

I think yes. It think it would help for CONNECT/OPTIONS/TRACE as well, but I have not run into servers which don't accept them there.

HEAD is a little different as it needs to pass the content length it would have had if it was a GET call.